pubsub/awssnssqs/awssnssqs.go

```diff
@@ -678,15 +678,14 @@ func (t *sqsTopic) SendBatch(ctx context.Context, dms []*driver.Message) error {
 			if len(attrs) == 0 {
 				attrs = nil
 			}
-			entry := sqstypesv2.SendMessageBatchRequestEntry{
+			entry := &sqstypesv2.SendMessageBatchRequestEntry{
 				Id:                aws.String(strconv.Itoa(len(req.Entries))),
 				MessageAttributes: attrs,
 				MessageBody:       aws.String(body),
 			}
-			req.Entries = append(req.Entries, entry)
 			if dm.BeforeSend != nil {
 				asFunc := func(i interface{}) bool {
-					if p, ok := i.(*sqstypesv2.SendMessageBatchRequestEntry); ok {
+					if p, ok := i.(**sqstypesv2.SendMessageBatchRequestEntry); ok {
```

Comment:
@vangent Will this break existing code that was using the BeforeSend to modify contents of MessageAttributes rather than needing to modify the fields such as MessageGroupId or DeduplicationId? And does it need to handle both cases with a warning on it initially?

e.g.:
```
	if p, ok := i.(**sqstypesv2.SendMessageBatchRequestEntry); ok {
		*p = entry
		return true
	}
	if p, ok := i.(*sqstypesv2.SendMessageBatchRequestEntry); ok {
		*p = *entry
		// emit some warning here that the caller needs to switch as it's getting a copy with
		// some fields being up-datable as they are copies of references
		return true
	}
```
Or will it be acceptable to wait and see if anyone has successfully used the existing code without needing to update top level fields directly when using V2 and then band-aid it?

Asking in case this causes a surprise for someone else after the next release (I work with @szaher)

Reply:
Yes, I think this will break existing code. The `asFunc` will fail to convert to the old non-pointer type.

I will include a note on this in the release notes, I'm not sure what else to do. There probably aren't too many users of this, as V2 support is relatively recent (and use of `As` is also uncommon).